src/test/clojure/xtdb/remote_test.clj

```diff
@@ -13,3 +20,198 @@
+                 set))
+          "testing query")))
+
+(deftest json-repsonse-test
+  (xt/submit-tx *node* [(xt/put :foo {:xt/id 1})])
+  (Thread/sleep 100)
+
+  (t/is (= {"latest-completed-tx"
+            {"tx-id" 0, "system-time" "2020-01-01T00:00:00Z"},
+            "latest-submitted-tx"
+            {"tx-id" 0, "system-time" "2020-01-01T00:00:00Z"}}
+           (-> (http/request {:accept :json
+                              :as :string
+                              :request-method :get
+                              :url (http-url "status")})
+               :body
+               decode-json))
+        "testing status")
+
+  (t/is (= {"tx-id" 1, "system-time" "2020-01-02T00:00:00Z"}
+           (-> (http/request {:accept :json
+                              :as :string
+                              :request-method :post
+                              :content-type :transit+json
+                              :form-params {:tx-ops possible-tx-ops}
+                              :transit-opts xtc/transit-opts
+                              :url (http-url "tx")})
+               :body
+               decode-json))
+        "testing tx")
+
+  (Thread/sleep 100)
+  (t/is (= [{:xt/id 1}]
+           (xt/q *node* '(from :docs [xt/id]))))
+
+  (let [tx (xt/submit-tx *node* [(xt/put :docs {:xt/id 2})])]
+    (t/is (= #{{"id" 2} {"id" 1}}
+             (-> (http/request {:accept "application/jsonl"
+                                :as :string
+                                :request-method :post
+                                :content-type :transit+json
+                                :form-params {:query '(from :docs [xt/id])
+                                              :basis {:tx tx}}
+                                :transit-opts xtc/transit-opts
+                                :url (http-url "query")})
+                 :body
+                 decode-json*
+                 set))
+          "testing query"))
+
+  (t/is (= {:status 400,
+            :body
+            {"xtdb.error/error-type" "illegal-argument",
+             "xtdb.error/error-key" "xtql/malformed-table",
+             "xtdb.error/message" "Illegal argument: ':xtql/malformed-table'",
+             "table" "docs",
+             "from" ["from" "docs" ["name"]]}}
+           (-> (http/request {:accept "application/jsonl"
+                              :as :string
+                              :request-method :post
+                              :content-type :transit+json
+                              :form-params {:query '(from docs [name])}
+                              ;; :transit-opts xtc/transit-opts
+                              :url (http-url "query")
+                              :throw-exceptions? false}
+                             #_#_identity identity)
+               (update :body decode-json)
+               (select-keys [:status :body])))
+        "illegal argument error")
+
+  ;; TODO how to mark this as an error
+  (t/is (= {:status 200,
```

Comment:
this should likely return a 400 status, if it doesn't already

Reply:
We are writing the result set to an output stream, when we are hitting this error we might have already written results to the output stream. I did this to replicate the exact behavior of an in memory node. So our client currently only throws when it actually consumes this error from the result set. So in transit you get something like 
```clj
{:foo "bar} ;; row 1
{:foo "bar} ;; row 2
#xt/runtime-error {.....}
```
and only when one consumes the 3rd row the client throws.